toolchain/lower/function_context.cpp

```diff
@@ -217,36 +217,60 @@ auto FunctionContext::GetDebugLoc(SemIR::InstId inst_id) -> llvm::DebugLoc {
                                loc.column_number, di_subprogram_);
 }
 
-auto FunctionContext::FinishInit(SemIR::TypeId type_id, SemIR::InstId dest_id,
+auto FunctionContext::FinishInit(TypeInFile type, SemIR::InstId dest_id,
                                  SemIR::InstId source_id) -> void {
-  switch (SemIR::InitRepr::ForType(sem_ir(), type_id).kind) {
+  switch (GetInitRepr(type).kind) {
     case SemIR::InitRepr::None:
       break;
     case SemIR::InitRepr::InPlace:
       if (sem_ir().constant_values().Get(source_id).is_constant()) {
         // When initializing from a constant, emission of the source doesn't
         // initialize the destination. Copy the constant value instead.
-        CopyValue(type_id, source_id, dest_id);
+        CopyValue(type, source_id, dest_id);
       }
       break;
     case SemIR::InitRepr::ByCopy:
-      CopyValue(type_id, source_id, dest_id);
+      CopyValue(type, source_id, dest_id);
       break;
     case SemIR::InitRepr::Incomplete:
       CARBON_FATAL("Lowering aggregate initialization of incomplete type {0}",
-                   sem_ir().types().GetAsInst(type_id));
+                   type.file->types().GetAsInst(type.type_id));
   }
 }
 
-auto FunctionContext::GetTypeIdOfInstInSpecific(SemIR::InstId inst_id)
-    -> std::pair<const SemIR::File*, SemIR::TypeId> {
-  return SemIR::GetTypeOfInstInSpecific(specific_sem_ir(), specific_id(),
-                                        sem_ir(), inst_id);
+auto FunctionContext::GetTypeIdOfInst(SemIR::InstId inst_id) -> TypeInFile {
+  auto [file, type_id] = SemIR::GetTypeOfInstInSpecific(
+      specific_sem_ir(), specific_id(), sem_ir(), inst_id);
+  return {.file = file, .type_id = type_id};
 }
 
-auto FunctionContext::CopyValue(SemIR::TypeId type_id, SemIR::InstId source_id,
+auto FunctionContext::GetValueRepr(TypeInFile type) -> ValueReprInFile {
+  auto result = ValueReprInFile{
```

Comment:
As long as we're spelling the type, we might as well put it in the usual position.

```suggestion
  ValueReprInFile result = {
```

Reply:
Ironically putting the type in the usual position is unusual in the toolchain :) Done.